packages/flutter/test/cupertino/theme_test.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+int timesBuilt;
+CupertinoThemeData actualTheme;
+
+final Widget child = Builder(
```

Comment:
The way the tests are set up seems a little clunky.

Maybe dispense with all of the global variables and just collect the information needed for each test like this:

```
Widget buildFrame(CupertinoThemeData theme,  WidgetBuilder onBuild) {
  return CupertinoTheme(
    data: theme,
    child: Builder(
      builder: (BuildContext context) {
        onBuild(context);
        return const Placeholder();
      }
    ),
  );
}

...

testWidgets('Default theme has defaults', (WidgetTester tester) async {
  CupertinoTheme theme;
  await tester.pumpWidget(
    buildFrame(
      const CupertinoThemeData(),
      (BuildContext context) { theme = CupertinoTheme.of(context); },
    ),
  );
  expect(theme, isNotNull);
  expect(theme.brightness, Brightness.light);
  ...
}
```

Or carry on with the global variables but expose all of the values (actualTheme and buildCount) via global variables

```
int buildCount = 0;
CupertinoThemeData actualTheme;

Widget buildFrame(CupertinoThemeData theme) {
  return CupertinoTheme(
    data: theme,
    child: Builder(
      builder: (BuildContext context) {
        actualTheme = CupertinoTheme.of(context);
        buildCount += 1;
        return const Placeholder();
      }
    ),
  );
}

...

testWidgets('Default theme has defaults', (WidgetTester tester) async {
  await tester.pumpWidget(buildFrame(const CupertinoThemeData));
  expect(actualTheme, isNotNull);
  expect(actualTheme.brightness, Brightness.light);
  expect(actualTheme.primaryColor, CupertinoColors.activeBlue);
  expect(actualTheme.textTheme.textStyle.fontSize, 17.0);
});
```

Reply:
The key there is to decouple the CupertinoTheme building (which happens frequently in the tests) from its child building (which should only happen at the beginning of the test and then only subsequently on rebuilds triggered by the inherited model). 

The inner builder can't be part of the buildFrame. I need to pass in a specific widget instance so calling buildFrame doesn't cause the inner child to build.

Reply:
Got it, that makes sense.  Since this is a little tricky, a comment about what's going on would help. And using a more specific var name than "child" would be less likely to cause a confusing error later.

Reply:
ah ya, the name child gets easily shadowed too. Changed to singletonThemeSubtree here and in material test